### PR TITLE
Implemented Switchable leg orientation

### PR DESCRIPTION
**Pull request by Maaz0070**

### CODE REVIEW

File: Drive/rover_drive_system.hpp [hunk 1]
```diff
@@ -408,11 +420,13 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
   const float kMaxTurnRadius  = 45;
   const float kLerpStep       = 0.5;
 
+
  public:
   MissionControlData mc_data_;
   Wheel & left_wheel_;
   Wheel & right_wheel_;
   Wheel & back_wheel_;
+  Wheel arrays[3] = {left_wheel_, right_wheel_, back_wheel_};
```
Comment (naterpotatoers) [inline Drive/rover_drive_system.hpp]: Was this utilized anywhere?

**Review by naterpotatoers — CHANGES_REQUESTED** [review]
@kammce Do you have any ideas for resolving SJSURoboticsTeam/urc-central-2021/issues/167

File: Drive/rover_drive_system.hpp (hunk 1 above)
Comment (Maaz0070) [reply Drive/rover_drive_system.hpp]: No. I Forgot to remove it.

**Comment by naterpotatoers** [comment]
Which files did you directly modify from Dev2? We only need the files you modified. Everything else can be imported from dev2 still

**Comment by Maaz0070** [comment]
> 
Since the files I modified included other files, mainly in the util folder, I just added the util folder. Everything outside the util folder are files I modified. 
Ill change it to where we don't import the entire util folder and only the ones we need.

File: Drive/rover_drive_system.hpp [hunk 2]
```diff
@@ -43,10 +43,28 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
...
     right_wheel_.Initialize();
     back_wheel_.Initialize();
+    
+
     SetSpinMode();
     sjsu::LogInfo("Drive system initialized!");
   };
 
+  // example Call: drive.switchLegOrientation(drive, back_wheel, right_wheel, back_wheel)
+  void switchLegOrientation(RoverDriveSystem &drive, Wheel &left, Wheel &right, Wheel &back)
```
Comment (naterpotatoers) [inline Drive/rover_drive_system.hpp]: Possible Wheel Orientations
Left, Right, Back --> Back, Left, Right --> Right, Back, Left
Back, Left, Right 
Right, Back, Left  
Not Possible Wheel Orientations (inverts drive steer logic)
Left, Back, Right
Right, Right, Right
```suggestion
  void SwitchLegOrientation(RoverDriveSystem &drive, Wheel &left, Wheel &right, Wheel &back)
```

**Comment by naterpotatoers** [comment]
As per the Discord thread 'Updating Libraries' we can try inheriting the RMD, Gpio, and CAN drivers and adding in our logic changes that way. Also good idea to just rename these drivers to something different

File: Drive/rover_drive_system.hpp [hunk 3]
```diff
@@ -41,20 +42,43 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
...
 
+  // example Call: drive.switchLegOrientation(drive, back_wheel, right_wheel, back_wheel)
+  void switchLegOrientation(int position)
+  {
+    /*
+    [0] = left right back
+    [1] = right back left
+    [2] = back left right
+    */
+    if(left_wheel_->GetHubSpeed() == 0 && right_wheel_->GetHubSpeed() == 0 && back_wheel_->GetHubSpeed() == 0)
```
Comment (naterpotatoers) [inline Drive/rover_drive_system.hpp]:
```suggestion
    if(IsStopped())
```
We have a helper function for this call

File: Drive/rover_drive_system.hpp [hunk 4]
```diff
@@ -41,20 +42,43 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
...
+    {
+        left_wheel_= wheels[(position + 0) % 3];
+        right_wheel_ = wheels[(position + 1) % 3];
+        back_wheel_ = wheels[(position + 2) % 3];
+    }
+    else
+    {
+      sjsu::LogError("Cannot change leg orientation while wheels are moving!");
+    }
```
Comment (naterpotatoers) [inline Drive/rover_drive_system.hpp]: This should never be ran. Our code should be smart enough to stop itself before this function ever gets called imo

File: Drive/rover_drive_system.hpp [hunk 5]
```diff
@@ -41,20 +42,43 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
...
+    right_wheel_->Initialize();
+    back_wheel_->Initialize();
+    
+
     SetSpinMode();
     sjsu::LogInfo("Drive system initialized!");
   }
 
+  // example Call: drive.switchLegOrientation(drive, back_wheel, right_wheel, back_wheel)
+  void switchLegOrientation(int position)
```
Comment (naterpotatoers) [inline Drive/rover_drive_system.hpp]:
```suggestion
  void SwitchLegOrientation(int position)
```

**Review by naterpotatoers — CHANGES_REQUESTED** [review]
Make sure to fix all the unit tests for drive system so that they all pass

**Comment by naterpotatoers** [comment]
Is the PR good to close since other PR got merged with these changes?

**Comment by Maaz0070** [comment]
> Is the PR good to close since other PR got merged with these changes?

Yes

**Pull request closed without merging**